widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx

```diff
@@ -112,6 +118,16 @@ export function ConfirmWalletsModal(props: PropTypes) {
     lastStepToBlockchainMeta &&
     !isValidAddress(lastStepToBlockchainMeta, destination);
 
+  const resetStates = () => {
```

Comment:
state shouldn't be reset manually. it breaks easily in future.
You can write an higher order function to get `open` and after finishing the exit scenario pass it to actual Modal. As an example when `open` set to `true`, you HOC will get the value and keeps it, then after finishing the exit animation, then you will pass the `open` value to Modal eventually.

Reply:
Added a component named "ConditionalMountComponent" to ui repository which is able to mount and unmount its children components with a delay.